test/from-draft3/to-draft4/references-tweak.json

```diff
@@ -0,0 +1,89 @@
+[
+    {
+        "title": "Reference to `type` when it has objects in its array",
+        "from": {
+            "type": [ {} ],
+            "$ref": "#/type/0"
+        },
+        "to": {
+            "$ref": "#/anyOf/0",
+            "anyOf": [
+                {}
+            ]
+        }
+    },
+    {
+        "title": "Reference to `type` when it has objects and string in its array",
+        "from": {
+            "type": [ "string", {} ],
+            "$ref": "#/type/1"
+        },
+        "to": {
+            "anyOf": [
+                { "type": ["string"] },
+                {}
+            ]
+        }
+    },
+    {
+        "title":"Reference to `disallow` when it has objects in its array",
+        "from": {
+            "disallow": [ {} ],
+            "$ref": "#/disallow/0"
+        },
+        "to": {
+            "allOf": [
+                { "not": {} }
+            ],
+            "$ref": "#/allOf/0/not"
+        }
+    },
+    {
+        "title": "Reference to `disallow` when it has objects and string in its array",
+        "from": {
+            "disallow": [ "string", {} ],
+            "$ref": "#/disallow/1"
+        },
+        "to": {
+            "allOf": [
```

Comment:
It is valid, but shouldn't the subschemas in `allOf` be in the opposite order? i.e. the `disallow` keyword first disallows `string`, so shouldn't the type string `not` be first in the array?

Reply:
Yeah! by doing it this way I can get the indexes of objects, like 
If I choose to upgrade index 0: `string` ... Then I will lose track of the index where the object was present.
If `string` is popped first, then array will be `[{}]`, then I lose track that it was in index 1, because now its in index 0